HW1/P6/P6.py

```diff
@@ -0,0 +1,112 @@
+import findspark
+findspark.init()
+import pyspark
+
+# shut down the previous spark context
+#sc.stop() 
+sc = pyspark.SparkContext()
+sc.setLogLevel('WARN')
+import numpy as np
+import matplotlib.pyplot as plt 
+import matplotlib.cm as cm
+import pdb
+import time
+import pandas as pd
+import itertools
+import itertools as it
+
+# Load the words from the file and extract them
+lines = sc.textFile('Shakespeare.txt')
+words = lines.flatMap(lambda line: line.split()).map(lambda x : x.encode('utf-8') ).map(lambda w: (w,1))
+
+# Filter the digits from the text
+words = words.filter(lambda w: False if w[0].isdigit() else True )
+
+# Filter the words entirely capitalize or entirely capitalize and end with a period
+words = words.filter(lambda w: False if w[0].isupper() else True ).keys()
+words_index = words.zipWithIndex()
+index_words = words_index.map(lambda (word, ind ): (ind,word) )
+
+time_start = time.time()
+final = []
+
+for phrase_number in range(0,10):
+    
+    print 'phrase number: ', phrase_number
+    print  'final: ', final
+    
+    if phrase_number==0:
+        word1 = 'He' ; word2 = 'can'
+    elif phrase_number==1:
+        word1 = 'While' ; word2 = 'he'
+    elif phrase_number==2:
+        word1 = 'To' ; word2 = 'be'
+    elif phrase_number==3:
+        word1 = 'Because' ; word2 = 'he'
+    elif phrase_number==4:
+        word1 = 'If' ; word2 = 'it'
+    elif phrase_number==5:
+        word1 = 'For' ; word2 = 'me'
+    elif phrase_number==6:
+        word1 = 'We' ; word2 = 'are'
+    elif phrase_number==7:
+        word1 = 'You' ; word2 = 'are'
+    elif phrase_number==8:
+        word1 = 'They' ; word2 = 'are'
+    elif phrase_number==9:
+        word1 = 'There' ; word2 = 'is'
+    
+    number_words = 2
+    phrase = [word1, word2]
+    while number_words <= 19: 
```

Comment:
This section of the code could be cleaner if you had a single rdd with pairs of words as keys so you only had to do a single lookup on the pair of words, not two lookups with one lookup for each word